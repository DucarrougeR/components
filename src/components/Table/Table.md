```js noeditor
const StatusAndResources = require('../../../styleguide_components/StatusAndResources').StatusAndResources;

<StatusAndResources
  status="experimental"
  figmaURL="https://www.figma.com/file/TIVgrnhEWwRGvr5yEQA4CVlj/Tables"
  githubURL="https://github.com/looker/lens/blob/master/src/components/Table/Table.tsx"
  feedbackTitle="Table Component Feedback" />
```


### Displaying tabular data

The `<Table>` component defines a structure for the display of tabular data. A `<TableHeader>` is used to display the table header and the `<TableBody>` is used to show the data rows.

```js
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell>Header 1</TableHeaderCell>
      <TableHeaderCell>Header 2</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell>Cell 1</TableDataCell>
      <TableDataCell>Cell 2</TableDataCell>
    </TableRow>
  </TableBody>
</Table>
```


