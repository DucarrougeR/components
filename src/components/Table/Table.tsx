import styled from '../../styled_components'
import { reset } from '../../styles/helpers'

export const Table = styled.table`
  ${reset};
  width: 100%;
  border-collapse: collapse;
`
