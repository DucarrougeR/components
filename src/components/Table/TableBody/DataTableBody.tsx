import * as React from 'react'
const classNames = require('classnames')
import * as styles from './DataTableBody.scss'

export interface DataTableBodyProps {
  align?: 'top' | 'middle' | 'bottom' | undefined
  textAlign?: 'left' | 'center' | 'right' | undefined
  className?: string | undefined
}

interface PropertyBag {
  [key: string]: any
}

export const DataTableBody: React.SFC<DataTableBodyProps> = ({
  className,
  align,
  textAlign,
  ...args
}) => {
  const styleableProps: PropertyBag = {
    [styles.alignMiddle]: align === 'middle',
    [styles.alignBottom]: align === 'bottom',
    [styles.textAlignLeft]: textAlign === 'left',
    [styles.textAlignRight]: textAlign === 'right',
    [styles.textAlignCenter]: textAlign === 'center'
  }

  return (
    <tbody
      className={classNames(
        styles.lensDataTableBody,
        className,
        styleableProps
      )}
      {...args}
    >
      {args.children}
    </tbody>
  )
}
