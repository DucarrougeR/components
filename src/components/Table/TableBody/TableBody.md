### Table Body

This component is used to define TableHeaderCelle body of the table and holds the data for the table. The `<TableBody>` can have multiple `<TableRow>` tags for each row of data, and a `<TableDataCell>` tag for each cell of data to be semantically correct.

```js
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell>Column 1</TableHeaderCell>
      <TableHeaderCell>Column 2</TableHeaderCell>
      <TableHeaderCell>Column 3</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
  </TableBody>
</Table>
```
