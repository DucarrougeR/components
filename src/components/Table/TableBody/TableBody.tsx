import styled from '../../../styled_components'

export type TableBodyAlignment = 'bottom' | 'middle' | 'top'
export type TableTextAlignment = 'center' | 'left' | 'right'

export interface TableBodyProps {
  align?: TableBodyAlignment
  textAlign?: TableTextAlignment
}

export const TableBody = styled<TableBodyProps, 'tbody'>('tbody')`
  vertical-align: ${props => props.align || 'top'};
  text-align: ${props => props.textAlign || 'left'};
`
