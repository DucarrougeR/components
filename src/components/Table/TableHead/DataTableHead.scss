.lens-data-table-head {
  text-align: left;
}
