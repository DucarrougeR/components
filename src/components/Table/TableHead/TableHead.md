### Table Headings

This component is used to define the heading of the table and describes the rows of data below. The `<TableHeader>` should have one `<TableRow>` tag and then use the `<TableHeaderCell>` tags to be semantically correct.

```js
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell>Column 1</TableHeaderCell>
      <TableHeaderCell>Column 2</TableHeaderCell>
      <TableHeaderCell>Column 3</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
  </TableBody>
</Table>
```
