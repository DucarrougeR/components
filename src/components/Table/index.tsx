export { Table } from './Table'
export { TableBody } from './TableBody/TableBody'
export { TableDataCell } from './TableDataCell/TableDataCell'
export { TableHead } from './TableHead/TableHead'
export { TableHeaderCell } from './TableHeaderCell/TableHeaderCell'
export { TableRow } from './TableRow/TableRow'
