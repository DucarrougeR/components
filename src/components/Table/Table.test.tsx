import * as React from 'react'
import { Table } from './Table'
import { DataTableHead } from './TableHead/DataTableHead'
import { DataTableBody } from './TableBody/DataTableBody'
import { create } from 'react-test-renderer'

test('A Table should accept html table children', () => {
  const component = create(
    <Table>
      <DataTableHead>
        <tr>
          <th>Header 1</th>
          <th>Header 1</th>
        </tr>
      </DataTableHead>
      <DataTableBody>
        <tr>
          <td>Header 1</td>
          <td>Header 1</td>
        </tr>
      </DataTableBody>
    </Table>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
