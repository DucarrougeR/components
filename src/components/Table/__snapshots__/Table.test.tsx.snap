// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Table should render 1`] = `
.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  width: 100%;
}

.c0 {
  border-collapse: collapse;
}

<table
  className="c0 c1"
/>
`;
